File: app/src/main/java/com/sleticalboy/dailywork/util/UnicodeUtils.java
package com.sleticalboy.dailywork.util;

public class UnicodeUtils {

    /**
     * 字符串转换 unicode
     */
    public static String string2Unicode(String string) {
        StringBuilder unicode = new StringBuilder();
        for (int i = 0; i < string.length(); i++) {
            // 取出每一个字符
            char c = string.charAt(i);
            // 转换为unicode
            unicode.append("\\u").append(Integer.toHexString(c));
        }
        return unicode.toString();
    }

    /**
     * unicode 转字符串
     */
    public static String unicode2String(String unicode) {
        StringBuilder string = new StringBuilder();
        String[] hex = unicode.split("\\\\u");
        for (int i = 1; i < hex.length; i++) {
            // 转换出每一个代码点
            int data = Integer.parseInt(hex[i], 16);
            // 追加成string
            string.append((char) data);
        }
        return string.toString();
    }

    /**
     * string 转 char
     */
    public static char string2char(String string) {
        int integer = 0;
        for (int i = 0; i < string.length(); i++) {
            // 取出每一个字符
            char c = string.charAt(i);
            integer += c;
        }
        integer = integer % 95 + 32;
        return (char) integer;
    }
}

File: MainApp/src/main/java/com/binlee/learning/util/UnicodeUtils.kt
package com.binlee.learning.util

object UnicodeUtils {

  /**
   * 字符串转换 unicode
   */
  fun string2Unicode(string: String): String {
    val unicode = StringBuilder()
    for (i in 0 until string.length) {
      // 取出每一个字符
      val c = string[i]
      // 转换为unicode
      unicode.append("\\u").append(Integer.toHexString(c.code))
    }
    return unicode.toString()
  }

  /**
   * unicode 转字符串
   */
  fun unicode2String(unicode: String): String {
    val string = StringBuilder()
    val hex = unicode.split("\\\\u".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    for (i in 1 until hex.size) {
      // 转换出每一个代码点
      val data = Integer.parseInt(hex[i], 16)
      // 追加成string
      string.append(data.toChar())
    }
    return string.toString()
  }

  /**
   * string 转 char
   */
  fun string2char(string: String): Char {
    var integer = 0
    for (i in 0 until string.length) {
      // 取出每一个字符
      val c = string[i]
      integer += c.code
    }
    integer = integer % 95 + 32
    return integer.toChar()
  }
}
